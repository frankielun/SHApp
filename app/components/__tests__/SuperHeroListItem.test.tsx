import React from 'react';
import renderer from 'react-test-renderer';
import { SuperHeroListItem } from '..';
import MockHeroData from '../../mock/MockHeroData';

// Updated test case with a Link to a different address
it('renders correctly', () => {
    const tree = renderer
      .create(<SuperHeroListItem key={`otherHeros_1`} item={MockHeroData as any} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });