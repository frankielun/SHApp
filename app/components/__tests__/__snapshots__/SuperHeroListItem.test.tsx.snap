// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "position": "relative",
      "width": 250,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "height": 170,
          "width": 246,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="contain"
      source={
        Object {
          "priority": "normal",
          "uri": undefined,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    numberOfLines={2}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 20,
        "color": "white",
        "fontSize": 18,
        "fontWeight": "bold",
        "left": 0,
        "position": "absolute",
      }
    }
  />
</View>
`;
